### ISSUE TITLE
Move scenarios from pro to beta

### ISSUE THREAD

**Issue by mabijkerk** [quintel/etengine#1431]
In the PR for Heat improvements (https://github.com/quintel/etengine/pull/1386), a migration was run on an outdated nl2019 dataset. The dataset was fixed before deploying to pro. This means that the scenarios on pro have been migrated properly, but the scenarios on beta haven't. The goal therefore is to move the II3050v2 and IP2024 scenarios from pro to beta.

- [ ] II3050v2 scenarios for 2050
- [ ] II3050v2 scenarios for 2040
- [ ] IP2024 scenarios for 2035
- [ ] IP2024 scenarios for 2030
- [ ] KEV scenarios for 2030
- [ ] IP2024 scenarios for 2025

@kndehaan is this something you could pick up? We should aim to do so before beta gets to far ahead of pro again.

@noracato could you list the instructions on how to do so?

**Comment by noracato** [comment]
This is done, right @kndehaan? Can we close?

**Comment by kndehaan** [comment]
@noracato unfortunately not yet.. It turns out that I still don't get the same values from the featured scenarios on pro to beta. I think obtaining balanced values could be a piece of the puzzle. I get back to you on how to obtain balanced values from scenarios through the API.

**Comment by kndehaan** [comment]
I'm quite stuck here. I've updated the featured scenarios on beta with the scenario user_values, custom curves and custom orders. However, there's still a difference in the dashboard item values between the pro and beta. 

I added functionalities to scenario-tools to get the custom curves and custom orders (https://github.com/quintel/scenario-tools/pull/40), as well as balanced user values (still work in progress on `upload-improvements` branche). I get the correct output here. 

However, when trying to upload the same settings, the following don't seem to work:
- set custom orders (excluding heat network orders which is already working on production) through `upload_custom_order` in the ETM_API
- set balanced values through `update_inputs_balanced_values` in the ETM_API
The following DO work:
- set custom curves
- set user values (except for flh_xxx_user_curve

So the not-working components is something where @noracato or @louispt1 could have a look some time? I think something goes wrong in the upload_xxx functions, see the working branche `upload-improvements` (it's quite in draft so ignore all comments and unfinished functions).

However, when I manually copy paste the user values, balanced values and custom orders from the pro to the beta scenarios in ETEngine, and when the custom curves are all uploaded, I would expect the same scenario outcomes. Is there something that I'm missing? @noracato can you have a look at this..? We could also have a look together.

**Comment by noracato** [comment]
Sorry to hear @kndehaan! We'll do our best to unstuck you!

I think the balanced values are not processed very well by the API. But as they are auto balanced, passing the argument `autobalance: true` to the request where you set the sliders should do the trick as well. (At least hypothetically)

The custom order API did change end of last year. I remember updating it for heat networks indeed. Let's check if the other custom orders also need a similar update!

**Comment by mabijkerk** [comment]
This  may be challenging because of the hidden curves.

**Comment by kndehaan** [comment]
> This may be challenging because of the hidden curves, that only appear if you set include_internal to true.

This PR tackles the problem of obtaining also the internal curves with the `get_template_settings` module.

**Comment by kndehaan** [comment]
I've updated the featured scenarios on beta with the scenario settings on pro (slider settings, balanced values, custom orders, custom curves including internal / hidden curves). I think the missing hidden curves on beta were the problem why the scenario outcomes (dashboard items) didn't match with the scenarios. 

Now, for some of the featured scenarios the (default) dashboard items match precisely between pro and beta. For some scenarios, there's a difference of 0.1 percentage pont or 0.1 bln/yr difference between pro and beta. Could this be due to rounding differences? I find the difference not significant enough to point out each scenario here where and for which dashboard item this is the case. 

For the scenarios below there are > 0.1 differences in one or more dashboard items:
- II3050v2 National Leadership - 2050:  a 0.3 percentage point difference in Biomass import (and a 0.1 percentage point difference in Renewables)
- II3050v2 Decentral Iniatives 2050: a 0.2 percentage point difference in Biomass import

I can't explain the remaining slight differences between pro and beta. @mabijkerk @noracato if any of you have suggestions, please let me know! Otherwise, I would say that these slight differences are acceptable and feel that we can close this issue.

**Comment by mabijkerk** [comment]
@kndehaan for National Leadership, the difference in biomass demand comes from a difference in green gas demand. This in turn comes from a difference in the use of the HT backup burner for the collective heat network. The final demand for steam hot water is the same on both pro and beta, so it seems to be caused by heat network supply. When I remove the capacity for P2H, the difference in dashboard items disappears - though there is still a marginal difference in integral costs per flexible technology...

Long story short: it would be good to find out why the results are still different. However, for now, the pro and beta scenarios are some much alike that the remaining difference is marginal.

**Comment by mabijkerk** [comment]
I have renamed this issue. When I open the National Leadership scenario locally (on master, with beta db), it produces different (though again, very close) results from both pro and beta. This is strange to me, as I would expect at least beta and local to be the same. We should make sure that, for the same db and the same commits, pro, beta and local produce stable results.

**Comment by kndehaan** [comment]
The featured scenarios on pro and beta now show the same results!🔥
I still need to check the featured scenarios when running the model locally, with the beta and pro db (takes some time to import). If all is good, then this issue can finally be closed.

**Comment by kndehaan** [comment]
With running the local model with pro db, the featured scenarios don't seem to correspond to pro or beta at all. Of the following featured scenarios, the scenario IDs of the scenarios on pro cannot be found at all when trying to open them locally:

- All II3050v2 2040 and 2050 scenarios
- All IP2024 2025, 2030 and 2035 scenarios

Of the following scenarios, the scenario IDs can be found, but just open a blank scenario of the correct region and end year:
- All KEV 2030 and 2035 scenarios
- 100% sustainable 2030 (Urgenda)
- All Northern Ireland scenarios except Business as Usual

For all other featured scenarios, the results just are incorrect. Since we're reviewing our way of working with the dbs, I don't think we should put effort in aligning this, right..? @mabijkerk @noracato
